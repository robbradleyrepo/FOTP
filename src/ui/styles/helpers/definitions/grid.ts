import factory, { StyleHelperProps } from "../factory";

type GridStyleKeys =
  | "_alignContent"
  | "_alignItems"
  | "_alignSelf"
  | "_columnGap"
  | "_gap"
  | "_grid"
  | "_gridArea"
  | "_gridAutoColumns"
  | "_gridAutoFlow"
  | "_gridAutoRows"
  | "_gridColumn"
  | "_gridColumnEnd"
  | "_gridColumnStart"
  | "_gridRow"
  | "_gridRowEnd"
  | "_gridRowStart"
  | "_gridTemplate"
  | "_gridTemplateAreas"
  | "_gridTemplateColumns"
  | "_gridTemplateRows"
  | "_justifyContent"
  | "_justifyItems"
  | "_justifySelf"
  | "_placeContent"
  | "_placeItems"
  | "_placeSelf"
  | "_rowGap";

export type GridProps = StyleHelperProps<GridStyleKeys>;

const grid = factory<GridStyleKeys>([
  { prop: "_gap", properties: ["gap"] },
  { prop: "_columnGap", properties: ["columnGap"] },
  { prop: "_rowGap", properties: ["rowGap"] },
  { prop: "_grid", properties: ["grid"] },
  { prop: "_gridArea", properties: ["gridArea"] },
  { prop: "_gridAutoColumns", properties: ["gridAutoColumns"] },
  { prop: "_gridAutoFlow", properties: ["gridAutoFlow"] },
  { prop: "_gridAutoRows", properties: ["gridAutoRows"] },
  { prop: "_gridColumn", properties: ["gridColumn"] },
  { prop: "_gridColumnEnd", properties: ["gridColumnEnd"] },
  { prop: "_gridColumnStart", properties: ["gridColumnStart"] },
  { prop: "_gridRow", properties: ["gridRow"] },
  { prop: "_gridRowEnd", properties: ["gridRowEnd"] },
  { prop: "_gridRowStart", properties: ["gridRowStart"] },
  { prop: "_gridTemplate", properties: ["gridTemplate"] },
  { prop: "_gridTemplateAreas", properties: ["gridTemplateAreas"] },
  { prop: "_gridTemplateColumns", properties: ["gridTemplateColumns"] },
  { prop: "_gridTemplateRows", properties: ["gridTemplateRows"] },
  { prop: "_alignContent", properties: ["alignContent"] },
  { prop: "_alignItems", properties: ["alignItems"] },
  { prop: "_alignSelf", properties: ["alignSelf"] },
  { prop: "_justifyContent", properties: ["justifyContent"] },
  { prop: "_justifyItems", properties: ["justifyItems"] },
  { prop: "_justifySelf", properties: ["justifySelf"] },
  { prop: "_placeContent", properties: ["placeContent"] },
  { prop: "_placeItems", properties: ["placeItems"] },
  { prop: "_placeSelf", properties: ["placeSelf"] },
]);

export default grid;
