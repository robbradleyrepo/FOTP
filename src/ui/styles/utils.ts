import { color } from "./variables";

interface DataUriFromPathParams {
  fill?: string;
  path: string;
  viewBox?: string;
  [k: string]: number | string | undefined;
}

const dataUriFromPath = ({
  fill = color.text.dark.base,
  path,
  viewBox = "0 0 100 100",
  ...rest
}: DataUriFromPathParams) => {
  const attributes = Object.entries(rest)
    .map(([key, value]) => ` ${key}="${value}"`)
    .join("");

  return `data:image/svg+xml;charset=utf8,${encodeURIComponent(`<svg viewBox="${viewBox}" xmlns="http://www.w3.org/2000/svg"${attributes}><path d="${path}" fill="${fill}" /></svg>
`)}`;
};

export { dataUriFromPath };
