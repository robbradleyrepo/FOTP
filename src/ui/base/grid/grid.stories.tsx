import React from "react";

import { percentage, s } from "@/common/ui/utils";

import { Grid, Item } from "./index";

export default {
  component: Grid,
  subcomponents: { Item },
  title: "Layout/Grid",
};

export const Default = () => (
  <div
    css={s({
      button: {
        marginRight: "1rem",
      },
    })}
  >
    <Grid
      gx={(t) => t.spacing.md}
      gy={(t) => t.spacing.md}
      itemWidth={[percentage(1 / 4)]}
    >
      {["Item 1", "Item 2", "Item 3", "Item 4"].map((i) => (
        <Item key={i}>
          <div
            css={s({
              backgroundColor: "#ccc",
            })}
          >
            {i}
          </div>
        </Item>
      ))}
    </Grid>
  </div>
);

export const TwoColumn = () => (
  <div
    css={s({
      button: {
        marginRight: "1rem",
      },
    })}
  >
    <Grid
      gx={(t) => t.spacing.md}
      gy={(t) => t.spacing.md}
      itemWidth={[percentage(1 / 2)]}
    >
      {["Item 1", "Item 2", "Item 3", "Item 4"].map((i) => (
        <Item key={i}>
          <div
            css={s({
              backgroundColor: "#ccc",
            })}
          >
            {i}
          </div>
        </Item>
      ))}
    </Grid>
  </div>
);

export const ThreeColumn = () => (
  <div
    css={s({
      button: {
        marginRight: "1rem",
      },
    })}
  >
    <Grid
      gx={(t) => t.spacing.md}
      gy={(t) => t.spacing.md}
      itemWidth={[percentage(1 / 3)]}
    >
      {["Item 1", "Item 2", "Item 3", "Item 4"].map((i) => (
        <Item key={i}>
          <div
            css={s({
              backgroundColor: "#ccc",
            })}
          >
            {i}
          </div>
        </Item>
      ))}
    </Grid>
  </div>
);

export const Responsive = () => (
  <div
    css={s({
      button: {
        marginRight: "1rem",
      },
    })}
  >
    <Grid
      gx={(t) => t.spacing.md}
      gy={(t) => t.spacing.md}
      itemWidth={[percentage(1 / 1), percentage(1 / 2), percentage(1 / 3)]}
    >
      {["Item 1", "Item 2", "Item 3", "Item 4"].map((i) => (
        <Item key={i}>
          <div
            css={s({
              backgroundColor: "#ccc",
            })}
          >
            {i}
          </div>
        </Item>
      ))}
    </Grid>
  </div>
);
