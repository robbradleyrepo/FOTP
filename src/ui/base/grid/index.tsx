import React, { createContext, forwardRef, PropsWithChildren } from "react";

import { ComponentStyleProps, negate, s, useTheme } from "@/common/ui/utils";

import { ResponsiveCSSValue } from "../../styles/helpers";
import { ThemeEnhanced } from "../../styles/theme";

type StyleValueFn = (theme: ThemeEnhanced) => ResponsiveCSSValue;

const useThemedCSSValue = () => {
  const theme = useTheme();

  return (value: ResponsiveCSSValue | StyleValueFn): ResponsiveCSSValue =>
    typeof value === "function" ? value(theme) : value;
};

interface GridContextProps {
  gx?: ResponsiveCSSValue;
  gy?: ResponsiveCSSValue;
  itemWidth?: ResponsiveCSSValue | StyleValueFn;
}

const GridContext = createContext<GridContextProps>({});

interface ItemProps extends ComponentStyleProps {
  width?: ResponsiveCSSValue | StyleValueFn;
}

export const Item = forwardRef<HTMLLIElement, PropsWithChildren<ItemProps>>(
  function Item({ _css = {}, children, width }, ref) {
    const themedCSSValue = useThemedCSSValue();

    return (
      <GridContext.Consumer>
        {({ gx, gy, itemWidth }) => (
          <li
            css={s(_css, {
              paddingLeft: gx,
              paddingTop: gy,
              width: themedCSSValue(
                typeof width === "undefined" ? itemWidth : width
              ),
            })}
            ref={ref}
          >
            {children}
          </li>
        )}
      </GridContext.Consumer>
    );
  }
);

export type GridAlignment = "center" | "left" | "right";
export type GridDirection = "ltr" | "rtl";

export interface GridProps extends ComponentStyleProps {
  align?: GridAlignment;
  direction?: GridDirection;
  gx?: ResponsiveCSSValue | StyleValueFn;
  gy?: ResponsiveCSSValue | StyleValueFn;
  innerCss?: ComponentStyleProps["_css"];
  itemWidth?: ResponsiveCSSValue | StyleValueFn;
}

export const Grid = forwardRef<HTMLDivElement, PropsWithChildren<GridProps>>(
  function Grid(
    {
      _css = {},
      align = "left",
      children,
      direction = "ltr",
      gx: _gx,
      gy: _gy,
      innerCss = {},
      itemWidth,
    },
    ref
  ) {
    const themedCSSValue = useThemedCSSValue();

    const gx = themedCSSValue(_gx);
    const gy = themedCSSValue(_gy);

    const negativeGx = Array.isArray(gx) ? gx.map(negate) : negate(gx);
    const negativeGy = Array.isArray(gy) ? gy.map(negate) : negate(gy);

    const justification = {
      center: "center",
      left: direction === "ltr" ? "flex-start" : "flex-end",
      right: direction === "ltr" ? "flex-end" : "flex-start",
    };

    return (
      <GridContext.Provider value={{ gx, gy, itemWidth }}>
        <div ref={ref} css={s(_css)}>
          <ul
            css={s(
              {
                alignItems: "stretch",
                display: "flex",
                flexDirection: direction === "ltr" ? "row" : "row-reverse",
                flexWrap: "wrap",
                justifyContent: justification[align],
              },
              innerCss,
              {
                marginLeft: negativeGx,
                marginTop: negativeGy,
              }
            )}
          >
            {children}
          </ul>
        </div>
      </GridContext.Provider>
    );
  }
);
