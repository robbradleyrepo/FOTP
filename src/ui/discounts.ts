import { captureException } from "@sentry/nextjs";
import { CartStatus, useCart } from "@sss/ecommerce/cart";
import { useLocale } from "@sss/i18n";
import { useRouter } from "next/router";
import { FC, useEffect, useRef } from "react";

import { useToastController } from "./base/toast";

const DISCOUNT_PARAMS = [
  {
    parameter: "aic",
    transform: (value: string) => `FRIEND-${value}`,
  },
  {
    parameter: "d",
  },
  {
    parameter: "discount",
  },
];

const enUsResource = {
  success: "{{ code }} has been automatically applied to your cart",
};

export const DiscountListener: FC = () => {
  const { discountCode, setDiscountCode, status } = useCart();
  const { i18n, t } = useLocale();
  const hasSyncedRef = useRef(false);
  const { query } = useRouter();
  const toast = useToastController();

  i18n.addResourceBundle("en-US", "DiscountListener", enUsResource);

  useEffect(() => {
    if (hasSyncedRef.current || status === CartStatus.INITIALIZING) {
      return;
    }

    const disabledCodes =
      process.env.DISABLED_DISCOUNT_CODES?.split(",").map((str) =>
        str.trim()
      ) ?? [];

    if (discountCode && disabledCodes.includes(discountCode)) {
      setDiscountCode(null);
    }

    let code: string | null = process.env.DEFAULT_DISCOUNT_CODE ?? null;

    if (code && disabledCodes.includes(code)) {
      captureException(
        `"DEFAULT_DISCOUNT_CODE" contains disabled discount code "${code}"`
      );

      code = null;
    }

    for (const {
      parameter,
      transform = (value: string) => value,
    } of DISCOUNT_PARAMS) {
      const param = query[parameter];
      if (typeof param === "string") {
        const qsCode = transform(param);

        if (!disabledCodes.includes(qsCode)) {
          code = qsCode;

          break;
        }
      }
    }

    if (code && code !== discountCode) {
      setDiscountCode(code);

      toast.push({
        children: t("DiscountListener:success", { code }),
      });
    }

    hasSyncedRef.current = true;
  }, [discountCode, hasSyncedRef, query, setDiscountCode, status, t, toast]);

  return null;
};
