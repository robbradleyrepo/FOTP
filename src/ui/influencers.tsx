import { useApolloClient } from "@apollo/react-hooks";
import { CartStatus, useCart } from "@sss/ecommerce/cart";
import dynamic from "next/dynamic";
import { useRouter } from "next/router";
import React, { FC, useEffect, useRef, useState } from "react";

import {
  InfluencerRedirectData,
  TransformedInfluencerRedirect,
  transformInfluencerRedirect,
} from "../cms/influencer-redirect";
import { useModalController } from "./base/modal";

const InfluencerRedirectModal = dynamic(
  () => import("./modules/influencer-redirect-modal")
);

export const InfluencerListener: FC = () => {
  const client = useApolloClient();
  const cart = useCart();
  const hasSyncedRef = useRef(false);
  const { open, setIsOpen } = useModalController("influencer");
  const { query } = useRouter();
  const [
    influencerRedirect,
    setInfluencerRedirect,
  ] = useState<TransformedInfluencerRedirect | null>(null);

  useEffect(() => {
    if (
      hasSyncedRef.current ||
      cart.status === CartStatus.INITIALIZING ||
      typeof query.influencer !== "string"
    ) {
      return;
    }

    // We'll only attempt to sync once, even if we're not successful
    hasSyncedRef.current = true;

    (async () => {
      try {
        const { INFLUENCER_REDIRECT } = await import(
          "../cms/influencer-redirect"
        );

        const { data } = await client.query<InfluencerRedirectData>({
          query: INFLUENCER_REDIRECT,
          variables: { handle: query.influencer },
        });

        if (!data.influencerRedirect) return;

        const influencerRedirect = transformInfluencerRedirect(
          data.influencerRedirect
        );

        cart.setDiscountCode(influencerRedirect.discountCode);
        setInfluencerRedirect(influencerRedirect);
        setIsOpen(true);
      } catch (error) {
        // Fail silently
      }
    })();
  }, [cart, client, hasSyncedRef, query, setIsOpen]);

  return influencerRedirect ? (
    <InfluencerRedirectModal
      influencerRedirect={influencerRedirect}
      open={open}
      setIsOpen={setIsOpen}
    />
  ) : null;
};
