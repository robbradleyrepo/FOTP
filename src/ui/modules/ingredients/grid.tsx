import { useLocale } from "@sss/i18n";
import React, { FC, useState } from "react";

import { belt, mx, percentage, s, useTheme } from "@/common/ui/utils";

import { secondaryButton } from "../../base/button";
import { Grid, GridProps, Item } from "../../base/grid";
import IngredientsCard from "./card";
import type { CoreIngredientProperties } from "./data-mapping";

const enUsResource = {
  showMore: "View all {{ count }}",
};

export interface IngredientsGridProps extends GridProps {
  groupId?: string;
  ingredients: CoreIngredientProperties[];
  initialShowCount?: number;
}

const IngredientsGrid: FC<IngredientsGridProps> = ({
  _css = {},
  align,
  direction,
  gx,
  gy,
  ingredients,
  initialShowCount = 6,
  itemWidth = [
    percentage(1),
    null,
    percentage(1 / 2),
    percentage(1 / 3),
    percentage(1 / 4),
  ],
}) => {
  const { i18n, t } = useLocale();
  const [showMore, setShowMore] = useState(false);
  const theme = useTheme();

  i18n.addResourceBundle("en-US", "IngredientsGrid", enUsResource);

  return (
    <div css={s(belt, _css)}>
      <Grid
        _css={s(mx("auto"))}
        align={align}
        direction={direction}
        gx={gx ?? [theme.spacing.sm, null, theme.spacing.md]}
        gy={gy ?? [theme.spacing.sm, null, theme.spacing.md]}
        itemWidth={itemWidth}
      >
        {ingredients.map((ingredient, index) => (
          <Item
            key={ingredient.id}
            _css={s({
              display:
                showMore || index < initialShowCount
                  ? undefined
                  : ["none", null, "block"],
            })}
          >
            <IngredientsCard {...ingredient} />
          </Item>
        ))}
      </Grid>
      {!showMore && ingredients.length > initialShowCount && (
        <button
          css={s(secondaryButton(), (t) => ({
            display: ["block", null, "none"],
            marginTop: t.spacing.lg,
            ...mx("auto"),
          }))}
          onClick={() => setShowMore(true)}
        >
          {t("IngredientsGrid:showMore", { count: ingredients.length })}
        </button>
      )}
    </div>
  );
};

export default IngredientsGrid;
