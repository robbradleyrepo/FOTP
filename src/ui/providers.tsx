import { ApolloProvider } from "@apollo/react-hooks";
import { CustomerProvider } from "@sss/customer";
import { CartProvider } from "@sss/ecommerce/cart";
import { CartOfferProvider } from "@sss/ecommerce/offer/cart";
import { GorgiasController } from "@sss/gorgias";
import { Locale, LocaleProvider } from "@sss/i18n";
import { ApolloClient } from "apollo-boost";
import React, { FC } from "react";

import { ThemeProvider } from "@/common/ui/utils";

import { ModalController } from "./base/modal";
import { ToastController } from "./base/toast";
import { DiscountListener } from "./discounts";
import { InfluencerListener } from "./influencers";
import { MenuController } from "./nav/menu";
import { theme } from "./styles/theme";

interface PageProvidersProps {
  apolloClient: ApolloClient<any>; // eslint-disable-line @typescript-eslint/no-explicit-any
  locale: Locale;
}

const PageProviders: FC<PageProvidersProps> = ({
  apolloClient,
  children,
  locale,
}) => (
  <LocaleProvider locale={locale}>
    <ApolloProvider client={apolloClient}>
      <ThemeProvider theme={theme}>
        <CartProvider>
          <CartOfferProvider>
            <CustomerProvider>
              <ModalController>
                <GorgiasController>
                  <MenuController>
                    <ToastController>
                      {children}
                      <DiscountListener />
                      <InfluencerListener />
                    </ToastController>
                  </MenuController>
                </GorgiasController>
              </ModalController>
            </CustomerProvider>
          </CartOfferProvider>
        </CartProvider>
      </ThemeProvider>
    </ApolloProvider>
  </LocaleProvider>
);

export default PageProviders;
