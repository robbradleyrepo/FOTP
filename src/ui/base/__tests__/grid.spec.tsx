import React from "react";
import { create } from "react-test-renderer";
import { mocked } from "ts-jest/utils";

import * as styles from "@/common/ui/utils";

import { Grid, GridAlignment, GridDirection, Item } from "../grid";

const mockStyleComposer = mocked(styles.s);

beforeEach(() => {
  jest.clearAllMocks();
});

describe.skip("<Item />", () => {
  it("should use the `Grid` component's X-axis gutter values as its left padding", () => {
    create(
      <Grid gx="item left padding">
        <Item />
      </Grid>
    );

    expect(mockStyleComposer).toMatchSnapshot();
  });

  it("should use the `Grid` component's Y-axis gutter values as its top padding", () => {
    create(
      <Grid gy="item top padding">
        <Item />
      </Grid>
    );

    expect(mockStyleComposer).toMatchSnapshot();
  });

  it("should use the `Grid` component's `itemWidth` as the default width", () => {
    create(
      <Grid itemWidth="item default width">
        <Item />
      </Grid>
    );

    expect(mockStyleComposer).toMatchSnapshot();
  });

  it("should use its own `width` value in preference to the `Grid` component's `itemWidth`", () => {
    create(
      <Grid itemWidth="item default width">
        <Item width="item width" />
      </Grid>
    );

    expect(mockStyleComposer).toMatchSnapshot();
  });
});

describe.skip("<Grid />", () => {
  it("should use the negative X-axis gutter values as the `ul` element's left margin", () => {
    create(<Grid gx="grid negative left margin" />);

    expect(mockStyleComposer).toMatchSnapshot();
    mockStyleComposer.mockClear();
  });

  it("should use the negative Y-axis gutter values as the `ul` element's top margin", () => {
    create(<Grid gy="grid negative top margin" />);

    expect(mockStyleComposer).toMatchSnapshot();
    mockStyleComposer.mockClear();
  });

  it("should set the appropriate flexbox item justification on the `ul` element", () => {
    [
      { _justifyContent: "flex-start", align: "left" },
      { _justifyContent: "center", align: "center", direction: "ltr" },
      { _justifyContent: "flex-start", align: "left", direction: "ltr" },
      { _justifyContent: "flex-end", align: "right", direction: "ltr" },
      { _justifyContent: "center", align: "center", direction: "rtl" },
      { _justifyContent: "flex-end", align: "left", direction: "rtl" },
      { _justifyContent: "flex-start", align: "right", direction: "rtl" },
    ].forEach(({ _justifyContent, align, direction }) => {
      const renderer = create(
        <Grid
          align={align as GridAlignment}
          direction={direction as GridDirection}
        />
      );

      const { props } = renderer.root.findByProps({ as: "ul" });

      expect(props).toMatchObject({ _justifyContent });
    });
  });

  it("should set the appropriate flexbox direction on the `ul` element", () => {
    [
      { _flexDirection: "row" },
      { _flexDirection: "row", direction: "ltr" },
      { _flexDirection: "row-reverse", direction: "rtl" },
    ].forEach(({ _flexDirection, direction }) => {
      const renderer = create(<Grid direction={direction as GridDirection} />);

      const { props } = renderer.root.findByProps({ as: "ul" });

      expect(props).toMatchObject({ _flexDirection });
    });
  });
});
