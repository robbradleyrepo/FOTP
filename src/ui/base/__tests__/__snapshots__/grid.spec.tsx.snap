// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> should use the negative X-axis gutter values as the \`ul\` element's left margin 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {},
    ],
    Array [
      Object {},
      Object {
        "marginLeft": "-grid negative left margin",
        "marginTop": undefined,
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": [Function],
    },
    Object {
      "type": "return",
      "value": [Function],
    },
  ],
}
`;

exports[`<Grid /> should use the negative Y-axis gutter values as the \`ul\` element's top margin 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {},
    ],
    Array [
      Object {},
      Object {
        "marginLeft": undefined,
        "marginTop": "-grid negative top margin",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": [Function],
    },
    Object {
      "type": "return",
      "value": [Function],
    },
  ],
}
`;

exports[`<Item /> should use its own \`width\` value in preference to the \`Grid\` component's \`itemWidth\` 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {},
    ],
    Array [
      Object {},
      Object {
        "marginLeft": undefined,
        "marginTop": undefined,
      },
    ],
    Array [
      Object {},
      Object {
        "paddingLeft": undefined,
        "paddingTop": undefined,
        "width": "item width",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": [Function],
    },
    Object {
      "type": "return",
      "value": [Function],
    },
    Object {
      "type": "return",
      "value": [Function],
    },
  ],
}
`;

exports[`<Item /> should use the \`Grid\` component's \`itemWidth\` as the default width 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {},
    ],
    Array [
      Object {},
      Object {
        "marginLeft": undefined,
        "marginTop": undefined,
      },
    ],
    Array [
      Object {},
      Object {
        "paddingLeft": undefined,
        "paddingTop": undefined,
        "width": "item default width",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": [Function],
    },
    Object {
      "type": "return",
      "value": [Function],
    },
    Object {
      "type": "return",
      "value": [Function],
    },
  ],
}
`;

exports[`<Item /> should use the \`Grid\` component's X-axis gutter values as its left padding 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {},
    ],
    Array [
      Object {},
      Object {
        "marginLeft": "-item left padding",
        "marginTop": undefined,
      },
    ],
    Array [
      Object {},
      Object {
        "paddingLeft": "item left padding",
        "paddingTop": undefined,
        "width": undefined,
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": [Function],
    },
    Object {
      "type": "return",
      "value": [Function],
    },
    Object {
      "type": "return",
      "value": [Function],
    },
  ],
}
`;

exports[`<Item /> should use the \`Grid\` component's Y-axis gutter values as its top padding 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {},
    ],
    Array [
      Object {},
      Object {
        "marginLeft": undefined,
        "marginTop": "-item top padding",
      },
    ],
    Array [
      Object {},
      Object {
        "paddingLeft": undefined,
        "paddingTop": "item top padding",
        "width": undefined,
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": [Function],
    },
    Object {
      "type": "return",
      "value": [Function],
    },
    Object {
      "type": "return",
      "value": [Function],
    },
  ],
}
`;
